plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -190,39 +193,33 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
       void send() {
          if (sending)
             return;
-         if (send_queue.empty())
-            return send_update();
+         if (send_queue.empty()) {
+            app().post(priority::medium, [self = this->shared_from_this()]() {
+               self->send_update();
+            });
+            return;
+         }
          sending = true;
-         socket_stream.binary(sent_abi);
-         sent_abi = true;
-         socket_stream.async_write(             //
-             boost::asio::buffer(send_queue[0]), //
+
+         socket_stream.async_write(
+             boost::asio::buffer(send_queue[0]),
              [self = this->shared_from_this()](boost::system::error_code ec, size_t) {
-                self->callback(ec, "async_write", [self] {
-                   self->send_queue.erase(self->send_queue.begin());
-                   self->sending = false;
-                   self->send();
-                });
+                self->send_queue.erase( self->send_queue.begin() );
+                self->sending = false;
+                self->callback(ec, "async_write", [self] { self->send(); });
              });
       }
 
-      void send(const char* s) {
-          boost::asio::post(this->plugin->work_strand, [self = this->shared_from_this(), str = s ]() {
-            self->send_queue.push_back({str, str + strlen(str)});
-            self->send();
-         });
-      }
-
       template <typename T>
       void send(T obj) {
-         boost::asio::post(this->plugin->work_strand, [self = this->shared_from_this(), obj = std::move(obj) ]() {
+         boost::asio::post(this->plugin->thread_pool.get_executor(), [self = this->shared_from_this(), obj = std::move(obj) ]() mutable {
```

Comment:
It's okay to lose the FIFO ordering here right? Initial look seems okay, except maybe for a pathological case: if a flood of `get_status_request_v0` are sent in, some of the responses may be sent out of order.